users/api_views.py

```diff
@@ -14,14 +14,39 @@
 from rest_framework_simplejwt.tokens import RefreshToken
 from rest_framework_simplejwt.views import TokenObtainPairView, TokenRefreshView
 
-from .models import User
-from .serializers import UserSerializer
+from .models import Profile, User, UserSettings
+from .serializers import ProfileSerializer, UserRegistrationSerializer, UserSerializer, UserSettingsSerializer
 from .tokens import account_activation_token
 
 
+class JWTAuthenticatedView:
```

Comment:
Isn't this automatically supported by REST framework?
![image](https://github.com/user-attachments/assets/0e6083c0-b12a-4e8d-adcc-1ea47580fe07)
At least authentication_classes is not needed, so only permission_classes should be used. In that case, I would change JWTAuthenticatedView to AuthenticatedView, move to utis.py file  and use in both, views and API views.